### PR TITLE
User profile revamp - 3541, 3736, 3735, 3737, 4266

### PR DESCRIPTION
**Pull request by vShyave**

### CODE REVIEW

File: project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  designation = ''
+  constructor(
+    public dialogRef: MatDialogRef<DesignationRequestComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private configService: ConfigurationsService,
+    private requestService: RequestService,
+    private matSnackBar: MatSnackBar
+  ) { }
+
+  ngOnInit() {
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.ts]: please remove unused functions also please add description for all functions

File: project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      firstName: (this.data.portalProfile.personalDetails) ? (this.data.portalProfile.personalDetails.firstname || '') : '',
+      email: (this.data.portalProfile.personalDetails) ? (this.data.portalProfile.personalDetails.primaryEmail || '') : '',
+      mobile: (this.data.portalProfile.personalDetails) ? (this.data.portalProfile.personalDetails.mobile || '') : '',
+    }
+
+    postData.updateFieldValues.push(data)
+    this.requestService.createPosition(postData)
+    .pipe(takeUntil(this.destroySubject$))
+    .subscribe((_res: any) => {
+      this.matSnackBar.open('Your designation request has been submitted successfully!')
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.ts:61]: hardcoded text should come from language files

File: project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div class="flex flex-col gap-4">
+    <div class="text-base font-bold text-center">Request to Add Designation</div>
+    <div class="flex flex-col gap-1">
+        <div class="text-sm font-bold required">Enter designation</div>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.html:4]: please mention jira ticket ids in pr title

File: project/ws/app/src/lib/routes/profile-v2/components/withdraw-request/withdraw-request.component.ts [hunk 4]
```diff
@@ -27,7 +35,19 @@ export class WithdrawRequestComponent implements OnInit, OnDestroy {
   }
 
   handleWithdrawRequest(): void {
-
+    this.data.approvalPendingFields.forEach((_obj: any) => {
+      this.userProfileService.withDrawRequest(this.configService.unMappedUser.id, _obj.wfId)
+      .pipe(takeUntil(this.destroySubject$))
+      .subscribe((_res: any) => {
+        this.matSnackBar.open('Withdrawn transfer request successfully!')
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/withdraw-request/withdraw-request.component.ts:42]: same as above

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts [hunk 5]
```diff
@@ -608,6 +652,17 @@ export class ProfileViewComponent implements OnInit, AfterViewInit, OnDestroy {
     })
   }
 
+  getInitials(): void {
+    if (this.currentUser.firstName) {
+      if (this.currentUser.firstName.split(' ').length > 1) {
+        const nameArr = this.currentUser.firstName.split(' ')
+        this.nameInitials = nameArr[0].charAt(0) + nameArr[1].charAt(0)
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts:659]: check for nameArr length first

Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts:659]: I'm checking it in the line 657

File: project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.ts (hunk 2 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.ts:61]: This I will do it, once I complete all the functionalities

File: project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.html (hunk 3 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/designation-request/designation-request.component.html:4]: Done

File: project/ws/app/src/lib/routes/profile-v2/components/withdraw-request/withdraw-request.component.ts (hunk 4 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/withdraw-request/withdraw-request.component.ts:42]: Done

**Pull request merged**